programs/mango-v4/src/instructions/token_conditional_swap_create.rs

```diff
@@ -0,0 +1,59 @@
+use anchor_lang::prelude::*;
+
+use crate::accounts_ix::*;
+use crate::error::*;
+use crate::logs::TokenConditionalSwapCreateLog;
+use crate::state::*;
+
+#[allow(clippy::too_many_arguments)]
+pub fn token_conditional_swap_create(
+    ctx: Context<TokenConditionalSwapCreate>,
+    token_conditional_swap: TokenConditionalSwap,
+) -> Result<()> {
+    let group = ctx.accounts.group.load()?;
+    require!(
+        group.is_ix_enabled(IxGate::TokenConditionalSwapCreate),
+        MangoError::IxIsDisabled
+    );
+
+    let now_ts: u64 = Clock::get()?.unix_timestamp.try_into().unwrap();
+    if token_conditional_swap.is_expired(now_ts) {
+        msg!("Already expired, ignoring");
+        return Ok(());
+    }
+
+    require_gte!(token_conditional_swap.price_premium_bps, 0);
+    require_gte!(token_conditional_swap.maker_fee_bps, 0);
+    require_gte!(token_conditional_swap.taker_fee_bps, 0);
```

Comment:
these checks don't actually help, because the setting is on the group, move later or remove